source/pool/contracts/Pool.sol

```diff
@@ -363,4 +368,34 @@ contract Pool is IPool, Ownable {
     bytes32 leaf = keccak256(abi.encodePacked(participant, score));
     return MerkleProof.verify(proof, root, leaf);
   }
-}
+
+/** @notice withdraw function that uses signature instead of claim
+  * @param signature signature from activate or any admin
+  * @param messageHash hash of token address, minimumAmount, recipient and nonce
+  * @param token address
+  * @param amount uint256
+  * @param recipient address
+  */
+  function withdrawWithSignature(
+    bytes memory signature,
+    bytes32 messageHash,
```

Comment:
Should take a `nonce` argument and calculate the hash in the function body (see below).

Reply:
``nonce`` added to the argument.

Reply:
Do we need the `messageHash` argument if we're generating it in the function body?

Reply:
We would need it for verification purpose? If a user enters the wrong amount or token address (intentionally or unintentionally) then the generated hash will not match with the ``messageHash``.

Reply:
Generally, signed message implementations take the individual parameters and the signature as arguments. The signature contains the message hash anyways. 
If a user is inputting incorrect arguments in a function, then you would want the transaction to fail.

Reply:
Ok, so I will remove the ``messageHash`` and ``require(messageHash == calculatedHash, "HASH_NOT_VERIFIED")`` as any wrong input will result in the ``ECDSA.recover`` transaction to fail, so there is no need to compare the hash. Am I understanding this correctly?

Reply:
Yup, that is correct.